### ISSUE TITLE
Password change by itself / Status: disconnected

### ISSUE THREAD

**Issue by sdalee** [ottoszika/node-red-contrib-ewelink#74]
Hi,

This is a marvelous plugin. I have been installed it this day,  but I have found two issues.
First: 
I will have to set the password at eWeLink Credentials window, almost every time if I change something in a node because it always changes something by itself.
eWeLink Credentials field often shows some kinds of code (eg.: 86356054.2fcbbb) instead of e-mail | region. 
Second: 
I add your every nodes and some nodes especially temperature, humidity, temperature/humidity are disconnected, and they sent the following msg:
StatusCodeError: 503 - "<html>\r\n<head><title>503 Service Temporarily Unavailable</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>503 Service Temporarily Unavailable</h1></center>\r\n<hr><center>openresty</center>\r\n</body>\r\n</html>\r\n"

**Comment by ottoszika** [comment]
Hello @sdalee,

The **first issue** shouldn't be a blocking one. Your password is there, but the label cannot be retrieved in edit mode because they are stored as credentials beside your flow (this prevents you from accidentally sharing your flow with your credentials), so an ID is shown up. Do you get an error after deploying your nodes?

The **second issue** seems to be a problem with the API itself. I have some questions in order to identify the problem:
1. Can you provide me your eWeLink region?
2. Did you get this error on other nodes?
3. Is this a temporary error? (Is there any working scenario on the nodes you mentioned)

Thanks,
Otto

**Comment by sdalee** [comment]
Hello Otto,

First issue:
I haven't added nodes to node-red and it has worked well since I wrote this post. 


Second:
After every deployment, other and other nodes were disconnected status.  I have made 2 screenshots.  The time difference was 10 sec between two deployments. If I took another screenshot, I would get the third result. I use EU region,

![sonoff-nodered1](https://user-images.githubusercontent.com/33203873/71402528-651cee00-262d-11ea-8d4b-0073274e7527.jpg)

![sonoff-nodered2](https://user-images.githubusercontent.com/33203873/71402552-70701980-262d-11ea-946a-9b55c7f08447.jpg)

**Comment by ndupont-net** [comment]
Hi, I've experienced a similat  behavior in my flow, when I add a eWeLink node, the credentials also show the node id (like your "86356054.2fcbbb") instead of the e-mail account, and deploying at that moment will disconnect all the eWeLink nodes.
What works for me is to edit the credentials, just open de box, validate, and deploy.

**Comment by sdalee** [comment]
I havan't change and deploy node-red flows for two days. My actual screenshot (after I tried to inject with any inject node (timestamp/on/off node):

![sonoff-nodered3](https://user-images.githubusercontent.com/33203873/71487456-a83cb400-281b-11ea-8296-2b798bb82cc5.jpg)

As you can see on the picture, if I injected a "connected" eWeLink node, the debug node gets the following error message: {"error":401,"msg":"Authentication error"}
if I injected a "disconnected" eWeLink node, the debug node gets nothing.

I hope this helps you with troubleshooting.

**Comment by ottoszika** [comment]
Hello guys,

Are you authenticated on eWeLink mobile application? Unfortunately multiple sessions are not supported. This was the reason of implementing a global connection. If you are not authenticated on other eWeLink applications, please let me know.
On the other hand, the 503 error seems to be a server-side problem (I can't reproduce this).

**Comment by E1cid** [comment]
After reading this issue i believe it is the same as the one i raised a minute ago. There seems to be a limit of six devices. Create more accounts to add more devices.

**Comment by sdalee** [comment]
> Hello guys,
> 
> Are you authenticated on eWeLink mobile application? Unfortunately multiple sessions are not supported. This was the reason of implementing a global connection. If you are not authenticated on other eWeLink applications, please let me know.
> On the other hand, the 503 error seems to be a server-side problem (I can't reproduce this).

Yes, I use the app. What do I need to do with the app, that it doesn't disturb the operation of your nodes?

**Comment by tyeth** [comment]
Create second account in ewelink, share all devices with that account then use that for NodeRed

**Comment by ojeysky** [comment]
My ewelink nodes are all indicating connected status on node-red but every changes on each of the nodes are not taking effect on the SONOFF devices, what could be the way to fix this?

**Comment by tyeth** [comment]
The ewelink api now requires a premium ewelink subscription

**Comment by E1cid** [comment]
it seems that you can query the api but not action. i.e. i can get/query temps, humidity, and device states, but can not  set an action or get device activity.

**Comment by tyeth** [comment]
I might have overstated the restriction. All I really heard was that to use
ifttt.com with ewelink you will now need a premium ewelink subscription and
it was hinted at affecting the ewelink api.

  Might be finally time to look at alternatives to ewelink like installing
esphome or tasmota to update those Sonoff devices

**Comment by ewidance** [comment]
Or maybe to use the "DIY API MODE" (http://developers.sonoff.tech/sonoff-diy-mode-api-protocol.html ) allowing to leave regular sonoff firmware while having capability to drive it without cloud connection. 

Pro: 
- it's reversible, you can still go back to regular Ewelink mode
- Restful
- Mdns discoverable

Cons : 
- Only available with some devices. 
- It is not probably as bullet proof as tasmota
- mdns ca be a pain on some networks
- At time of writing no nodered module supports API mode....
